File: 844_Backspace_string_compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        stack<char> s1,s2;
        for(char c: s)
            if(c!='#')
                s1.push(c);
            else{
                if(!s1.empty())
                    s1.pop();
            }
        for(char c: t)
            if(c!='#')
                s2.push(c);
            else{
                if(!s2.empty())
                    s2.pop();
            }
        while(!s1.empty() && !s2.empty()){
            char c1=s1.top();
            s1.pop();
            char c2=s2.top();
            s2.pop();
            if(c1!=c2)
                return false;
        }
        if(!s1.empty() || !s2.empty())
            return false;
        return true;
    }
};

File: 844_Backspace String Compare.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        Stack<Character> s1=new Stack<>();
        Stack<Character> s2=new Stack<>();
        char[] c1=s.toCharArray();
        char[] t1=t.toCharArray(); 
        for(Character c: c1){
            if(c!='#') s1.push(c);
            else{
                if(!s1.isEmpty()) s1.pop();
            }
        }
        for(Character c: t1){
            if(c!='#') s2.push(c);
            else{
                if(!s2.isEmpty()) s2.pop();
            }
        }
        while(!s1.isEmpty() && !s2.isEmpty()){
            char c=s1.peek();
            char k=s2.peek();
            s1.pop();
            s2.pop();
            if(c!=k)
                return false;
        }
        if(!s1.isEmpty() || !s2.isEmpty())
            return false;
        return true;
    }
}
